### PR TITLE
customcluster:support kubeImageRepo and cluster setting config

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
part of #128 

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: docs/content/en/docs/cluster-operator/vms-cluster-lifecycle.md [hunk 1]
```diff
@@ -129,11 +129,13 @@ If you just modify the IP and apply the manifest directly, you will get a kubern
...
 
-| parameters | setting-position                            |
-| :-----|:--------------------------------------------| 
-| KubeVersion | KubeadmControlPlane.Spec.Version            | 
-| PodCIDR | Cluster.Spec.ClusterNetwork.Pods.CidrBlocks | 
-| CNIPlugin | CustomCluster.CNI.Type                      | 
+| parameters    | setting-position                                                |
+|:--------------|:----------------------------------------------------------------| 
+| KubeVersion   | KubeadmControlPlane.Spec.Version                                | 
+| PodCIDR       | Cluster.Spec.ClusterNetwork.Pods.CidrBlocks                     |
```
Comment (zirain) [inline docs/content/en/docs/cluster-operator/vms-cluster-lifecycle.md:135]: can you update link these to api docs?

Comment (Xieql) [reply docs/content/en/docs/cluster-operator/vms-cluster-lifecycle.md:135]: I attempted to run "make gen-api-doc", but no new content was generated automatically .
The variable is not located in customcluster/custommachine, but in the third-party cluster/kcp.
Therefore, may be I need to manually write documentation to describe these variables？

**Review by hzxuzhonghu — APPROVED** [review]

**Comment by Xieql** [comment]
> Please give more explict description about what does this pr support, and donot use ambiguous words like `more config`

ok

**Review by zirain — APPROVED** [review]
/lgtm

**Pull request merged**